waku/node/waku_node.nim

```diff
@@ -280,6 +402,26 @@ proc publish*(node: WakuNode, topic: PubsubTopic, message: WakuMessage) {.async,
       hash=topic.digest(message).to0xHex(),
       publishTime=getNowInNanosecondTime()
 
+proc autoPublish*(node: WakuNode, message: WakuMessage) {.async, gcsafe.} =
+  ## Publish a `WakuMessage`.`WakuMessage` must contain a
+  ## `contentTopic` field.
+
+  if node.wakuRelay.isNil():
+    error "Invalid API call to `publish`. WakuRelay not mounted. Try `lightpush` instead."
+    return
+
+  let shard = getShard(message.contentTopic).valueOr:
+    error "Autosharding error: ", error=error
+    return
+
+  discard await node.wakuRelay.publish(shard, message)
```

Comment:
As far as possible, IMO we should try to avoid duplicate logic e.g. "two places to publish from". In other words, this could be an overload for the `publish*` function, which would allow API users to publish without specifying a pubsubTopic. If no `pubsubTopic`, the sharding logic determines the `pubsubTopic` and simply call `publish*()` (with pubsubTopic) - the only place we publish from. This way we don't have to remember to log in both places or duplicate any other logic. We've simply added a short indirection to populate the `pubsubTopic` in the publish request.

Reply:
True, I always forget overloading since it's not a thing in Rust.